openwisp_monitoring/check/classes/iperf.py

```diff
@@ -320,6 +287,46 @@ def check(self, store=True):
         device_connection.disconnect()
         return result
 
+    def _get_iperf_test_conditions(self):
+        """
+        Returns iperf check test conditions (rev_or_bidir, end_condition)
+        """
```

Comment:
This is better, but I meant to add a method which returns the commands

```python
def _get_check_command(self):
    # build tcp and udp commands
    return tcp_command, udp_command
```    


Then from `check` method, we would call this method

```python
def check(self):
     tcp_command, udp_command = self._get_check_command()
     # perform check
```

This will make the code of `check` method easy to follow.

Reply:
Sure, I'll update that :+1: